### PR TITLE
test: add integration tests for external scanner edge cases

### PR DESCRIPTION
**Pull request by sergio-sisternes-epam**
## Description

Adds 6 integration tests to `tests/integration/test_audit_external_scanners.py` covering edge cases found during post-merge E2E validation of #1579 (external SARIF scanner ingestion).

The existing test suite covered the happy path (critical/info exit codes, flag gating, option conflicts) but missed several error-path and output-format scenarios that the manual E2E run surfaced.

## Type of change

- [x] Maintenance / refactor

## New tests

| Test | Scenario |
|------|----------|
| `test_flag_on_warning_only_exits_2` | Warning-only SARIF produces exit 2 (gap between critical=1 and info=0) |
| `test_missing_sarif_file_gives_actionable_error` | Non-existent file path produces clear error, no traceback |
| `test_malformed_sarif_gives_actionable_error` | SARIF missing `runs` key fails closed with descriptive message |
| `test_skillspector_not_on_path_gives_actionable_error` | Binary absent: error suggests `--external sarif` fallback |
| `test_sarif_output_includes_external_findings` | `-f sarif` output preserves external findings in SARIF 2.1.0 |
| `test_markdown_output_includes_external_findings` | `-f markdown` renders external findings in table |

All tests are hermetic (no network, no vendor binary, no markers needed).

## Testing

- [x] Tested locally
- [x] All existing tests pass
- [x] Added tests for new functionality (if applicable)

## Spec conformance (OpenAPM v0.1)

- [x] N/A -- this PR does not change OpenAPM-observable behaviour.

### CODE REVIEW

**Review by sergio-sisternes-epam — COMMENTED** [review]
**Automated review (agent-merge check-in)**

All 13 tests pass locally; CI is fully green (Lint, Build & Test shards, CodeQL, Merge Gate, CLA). Lint and format checks clean.

**What was checked:**
- Tests are hermetic (no network, no vendor binary) -- no markers needed per convention.
- `shutil.which` patch in `test_skillspector_not_on_path_gives_actionable_error` works correctly: patches the module-level import used by the production code.
- No URL substring assertions (test convention compliant).
- Existing fixtures (`runner`, `_isolate_config`, `_inject_flag`, `_write_sarif`) are properly reused.

**Minor non-blocking observations (can be addressed in follow-up):**
- `test_malformed_sarif_gives_actionable_error`: the `or "failed" in result.output.lower()` fallback is broad -- tightening to assert just the specific message would improve signal.
- `test_flag_on_warning_only_exits_2`: `>= 1` could be `== 1` for tighter duplicate-ingestion coverage.
- The `shutil.which` patch is module-global rather than scoped to the skillspector module -- low risk today but could be narrowed.

**Verdict:** Ready to merge. No blocking findings.

File: tests/integration/test_audit_external_scanners.py [hunk 1]
```diff
@@ -144,3 +144,84 @@ def test_unknown_scanner_name(runner, monkeypatch, tmp_path):
...
+
+def test_missing_sarif_file_gives_actionable_error(runner, monkeypatch, tmp_path):
+    """A path that does not exist produces a clear, non-traceback message."""
+    _inject_flag(monkeypatch, enabled=True)
+    monkeypatch.chdir(tmp_path)
+    result = runner.invoke(
+        audit, ["--external", "sarif", "--external-sarif", str(tmp_path / "no.sarif")]
+    )
+    assert result.exit_code == 2
+    assert "not found" in result.output.lower()
```
[the remark this answers is not in the document]
Comment (sergio-sisternes-epam) [reply tests/integration/test_audit_external_scanners.py:172]: Added `assert "Traceback" not in result.output` to match the docstring's intent. Good catch.

File: tests/integration/test_audit_external_scanners.py [hunk 2]
```diff
@@ -144,3 +144,84 @@ def test_unknown_scanner_name(runner, monkeypatch, tmp_path):
...
+
+def test_malformed_sarif_gives_actionable_error(runner, monkeypatch, tmp_path):
+    """A SARIF file missing the 'runs' key fails closed with a clear message."""
+    _inject_flag(monkeypatch, enabled=True)
+    bad = tmp_path / "bad.sarif"
+    bad.write_text('{"version":"2.1.0"}', encoding="utf-8")
+    monkeypatch.chdir(tmp_path)
+    result = runner.invoke(audit, ["--external", "sarif", "--external-sarif", str(bad)])
+    assert result.exit_code == 2
+    assert "not a SARIF document" in result.output or "failed" in result.output.lower()
```
[the remark this answers is not in the document]
Comment (sergio-sisternes-epam) [reply tests/integration/test_audit_external_scanners.py]: Replaced the loose `or "failed" in result.output.lower()` fallback with specific assertions: `assert "not a SARIF" in result.output` and `assert "missing 'runs'" in result.output`. Also added `assert "Traceback" not in result.output`. The assertions now match the exact error message from `sarif_ingest.py`.

**Review by danielmeppiel — APPROVED** [review]
